lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -59,15 +60,54 @@ def build_cmd(src)
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
 
+        def compile_cpp(src)
+          cmd = build_cpp_cmd(src)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def compile_cpp_files(files)
+          cmd = build_cpp_files_cmd(files)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def build_cpp_cmd(src)
+          src_file = Tempfile.create(['scexec', ".cpp"])
+          File.write(src_file.path, src)
+          build_cpp_files(src_file.path)
+        end
+
+        def build_cpp_files_cmd(file_array)
+          cmd = [ mingw_bin ]
+          cmd << file_array.join(" ")
+          cmd << "-I #{INCLUDE_DIR}"
+          cmd << "-o #{outfile}"
+          cmd << "-shared" if outfile.ends_with?('.dll')
+
+          cmd << opt_lvl if OPTIMIZATION_FLAGS.include?(opt_lvl)
+          cmd << compile_options
+
+          link_opts = ['-Wl']
+          link_opts << '-s' if strip_syms
+          link_opts << link_options
+          link_opts << "-T#{link_script} " if link_script
+
+          cmd << link_opts.join(",")
+          cmd.join(" ")
+        end
+
+        def cleanup_files
           unless self.keep_src
+            src_file = "#{self.file_name}.c"
             File.delete(src_file) if File.exist?(src_file)
           end
 
           unless self.keep_exe
+            exe_file = "#{self.file_name}.exe"
             File.delete(exe_file) if File.exist?(exe_file)
           end
         rescue Errno::ENOENT
```

Comment:
You may want to consider extending this code to support cleaning up CPP files given your changes.